lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -354,26 +354,6 @@ private void sendDetachMessage(CompletionListener listener) throws AblyException
         }
     }
 
-    public void sync() throws AblyException {
```

Comment:
Could you please explain why we deleted this method?

Reply:
It's unused method : (

Reply:
But it's public method for the `Channel`, even if it's not mentioned in docs, somebody might've used it, and it will be breaking change

Reply:
Btw, I wanted to add, there was a discussion for `ably-java` where lots of methods are marked public which are not necessarily used.

Reply:
Oh, that's bad. Can we agree to add `@VisibleForTesting` annotation when we find such methods? Also we should prefer package-private methods over public to ease testing, in some rare cases we can access private fields using reflection, but making methods public for testing should be avoided in future.

Reply:
Seems this is an actual method which was in use sometime back. We can it's usage in `ably-js` 
https://github.com/ably/ably-js/blob/fdd5a566eafac0a7fae9592f890361a608794487/src/common/lib/client/realtimechannel.ts#L914C7-L914C7

Reply:
https://github.com/ably/ably-js/blob/fdd5a566eafac0a7fae9592f890361a608794487/src/common/lib/client/realtimechannel.ts#L914